wafer/schedule/models.py

```diff
@@ -12,17 +12,60 @@
 from wafer.talks.models import Talk
 
 
-@python_2_unicode_compatible
-class Day(models.Model):
-    """Days on which the conference will be held."""
-    date = models.DateField(null=True, blank=True)
+# Validation Helpers
+def includes(obj1, obj2):
+    """Test if the times for obj1 are completely included in
+       the times for obj2"""
+    if (obj2.end_time >= obj1.end_time and
+        obj2.get_start_time() <= obj1.get_start_time()):
+        return True
+    return False
+
+def overlap(obj1, obj2):
+    """Test if obj1 and obj2 overlap on times"""
+    # obj1 and obj2 are either both ScheduleBlocks or both slots
+    if (obj2.get_start_time() >= obj1.get_start_time() and
```

Comment:
Alternative implementation ideas:
* We could write `if (obj1.get_start_time() <= obj2.get_start_time() < obj1.end_time): return True` unless there is an odd reason this doesn't work as expected, or
* We could write simply `if (obj1.end_time <= obj2.get_start_time()): return False` if we're happy relying on `obj1.get_start_time() < obj1.end_time`? If we check these two conditions instead, I think we can avoid the call to `include(..)`?

Reply:
Noting that this was rewritten and simplified in 157fe90